pkg/token/token-util/token-util.go

```diff
@@ -0,0 +1,75 @@
+//(C) Copyright 2019 Hewlett Packard Enterprise Development LP
+
+package tokenutil
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"log"
+	"strings"
+
+	"gopkg.in/square/go-jose.v2"
+)
+
+// Token a jwt token format
+type Token struct {
+	Issuer           string `json:"iss"`
+	Subject          string `json:"sub"`
+	Expiry           int64  `json:"exp"`
+	IssuedAt         int64  `json:"iat"`
+	Type             string `json:"typ"`
+	Nonce            string `json:"nonce"`
+	AtHash           string `json:"at_hash"`
+	ClientID         string `json:"cid,omitempty"`
+	UserID           string `json:"uid,omitempty"`
+	TenantID         string `json:"tenantId"`
+	AuthorizedParty  string `json:"azp"`
+	KeycloakClientID string `json:"clientId"`
+	IsHPE            bool   `json:"isHPE"`
+}
+
+// DecodeAccessToken decodes the accessToken offline
+func DecodeAccessToken(rawToken string) (Token, error) {
+	_, err := jose.ParseSigned(rawToken)
+	if err != nil {
+		return Token{}, fmt.Errorf("oidc: malformed jwt: %v", err)
+	}
+
+	// Throw out tokens with invalid claims before trying to verify the token. This lets
+	// us do cheap checks before possibly re-syncing keys.
+	payload, err := parseJWT(rawToken)
+	if err != nil {
+		log.Fatal(fmt.Sprintf("oidc: malformed jwt: %v", err))
+		return Token{}, fmt.Errorf("oidc: malformed jwt: %v", err)
+	}
+	var token Token
+	if err := json.Unmarshal(payload, &token); err != nil {
+		log.Fatal(fmt.Sprintf("oidc: failed to unmarshal claims: %v", err))
+		return Token{}, fmt.Errorf("oidc: failed to unmarshal claims: %v", err)
+	}
+
+	if token.UserID != "" {
+		// Okta User token
```

Comment:
I think this reference to okta should be removed.

Reply:
The reference to Okta is now removed.